### PR TITLE
Add py-atlinter

### PR DESCRIPTION
**Pull request by Stannislav**
Repo: https://bbpgitlab.epfl.ch/project/proj101/atlas_interpolation

### CODE REVIEW

**Comment by Stannislav** [comment]
I'm not sure how to effectively debug this, could someone please help?

I ran several tests on the BB5, and this worked:

1. Installing `py-torch` by itself (+`cuda_arch` flag as per @pramodk)
    ```
    $ spack install py-torch cuda_arch=70
    ...
    [+] /gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-torch-1.7.0-v6adwi
    ```

2. After that the actual `py-atlinter` package installation introduced in this PR (that depends on `py-torch`):
    ```
    $ spack install py-atlinter
    ...
    ==> Installing py-atlinter-0.1.0-yf2gcwndojbu6b5zocb3eb4xvs4d5uzx
    ==> No binary for py-atlinter-0.1.0-yf2gcwndojbu6b5zocb3eb4xvs4d5uzx found: installing from source
    ==> py-atlinter: Executing phase: 'build'
    ==> py-atlinter: Executing phase: 'install'
    ...
    [+] /gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atlinter-0.1.0-yf2gcw
    ```

File: var/spack/repos/builtin/packages/py-atlinter/package.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools-scm', type='build')
+
+    depends_on('py-atldld@0.2.2', type='run')
+    depends_on('py-numpy', type='run')
+    depends_on('py-pillow', type='run')
+    depends_on('py-pytorch-fid', type='run')
+    depends_on('py-requests', type='run')
+    depends_on('py-torch+cuda', when='+cuda', type='run')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-atlinter/package.py:31]: No tensorflow?

**Comment by matz-e** [comment]
Oh Jenkins, retest this please

…cleaned out the build directory. All local attempts to reproduce this build `py-torch`…

File: var/spack/repos/builtin/packages/py-atlinter/package.py (hunk 1 above)
Comment (EmilieDel) [reply var/spack/repos/builtin/packages/py-atlinter/package.py:31]: No we don't have `tensorflow` as dependency for `atlinter`.

**Comment by matz-e** [comment]
Oh Jenkins, retest this please

**Comment by matz-e** [comment]
Oh Jenkins, retest this please

**Review by matz-e — APPROVED** [review]

**Comment by matz-e** [comment]
Once this thing is done building (looks good so far, no functional changes, just jumping through hoops), please test it with:
```
unset MODULEPATH
. /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/1295/config/modules.sh
module load unstable py-atlinter
```
Enjoy!

**Comment by Stannislav** [comment]
Tested locally as suggested by @matz-e above, everything works. Merging.

**Pull request merged**